test/Metadata/MetadataTest.php

```diff
@@ -0,0 +1,150 @@
+<?php
+/**
+ * Zend Framework (http://framework.zend.com/)
+ *
+ * @link      http://github.com/zendframework/zf2 for the canonical source repository
+ * @copyright Copyright (c) 2005-2015 Zend Technologies USA Inc. (http://www.zend.com)
+ * @license   http://framework.zend.com/license/new-bsd New BSD License
+ */
+
+namespace ZendTest\Db\Metadata;
+
+use Zend\Db\Adapter\Platform\AbstractPlatform;
+use Zend\Db\Adapter\Platform\Mysql as MysqlPlatform;
+use Zend\Db\Adapter\Platform\SqlServer as SqlServerPlatform;
+use Zend\Db\Adapter\Platform\Sqlite as SqlitePlatform;
+use Zend\Db\Adapter\Platform\Postgresql as PostgresqlPlatform;
+use Zend\Db\Adapter\Platform\Oracle as OraclePlatform;
+use Zend\Db\Metadata\Metadata;
+
+class MetadataTest extends \PHPUnit_Framework_TestCase
+{
+    /**
+     * @dataProvider createSourceFromAdapterDataProvider
+     *
+     * @param AbstractPlatform $platform
+     * @param string           $sourceExpectation
+     */
+    public function testTheConstructorSetsTheAdaptersAndTheSource($platform, $sourceExpectation)
+    {
+        /** @var \Zend\Db\Adapter\Adapter $adapterMock */
+        $adapterMock = $this->getMockBuilder('Zend\Db\Adapter\Adapter')
+                            ->disableOriginalConstructor()
+                            ->getMock();
+
+        $adapterMock->expects($this->once())
```

Comment:
Could you extract all the thing about `adapterMock` in a method called `createAdapterMockWithPlatform($platform)`

I think will improve test readiblity.